lib/ood_core/job/adapters/helper.rb

```diff
@@ -12,6 +12,31 @@ class Helper
         def self.bin_path(cmd, bin_default, bin_overrides)
           bin_overrides.fetch(cmd.to_s) { Pathname.new(bin_default.to_s).join(cmd.to_s).to_s }
         end
+        
+        # Gets a command that submits command on another host via ssh
+        # @param submit_host [String] where to submit the command
+        # @param cmd [String] the desired command to execute on another host
+        # @param cmd_args [Array] arguments to the command specified above
+        # @param strict_host_checking [Bool] whether to use strict_host_checking
+        # @param env [Hash] env variables to be set w/ssh
+        #
+        # @return cmd [String] command wrapped in ssh if submit_host is present
+        # @return args [Array] command arguments including ssh_flags and original command
+        def self.ssh_wrap(submit_host, cmd, cmd_args, strict_host_checking = true, env = {})
+          return cmd, cmd_args if submit_host.to_s.empty?
+
+          check_host = strict_host_checking ? "yes" : "no"
+          args = ['-o', 'BatchMode=yes', '-o', 'UserKnownHostsFile=/dev/null', '-o', "StrictHostKeyChecking=#{check_host}", "#{submit_host}"]
+
+          command = ""
+          env.each {|key, value| command += "export #{key}=#{value}; "} # exports env vars
+          command += cmd_args.unshift(cmd).join(' ')
+          args.push(command)
```

Comment:
This does the same thing:

```
args.push(env.map {|k,v| "export #{key}=#{value};" } + [cmd] + cmd_args).join(' '))
```

though `env.map {|k,v| "export #{key}=#{value};" }` should be a separate method

Reply:
Should a new method still be made for this with the last change? There are quite a few params in this method already, but everything in this command deals with ssh wrapping.